entity_api/src/query.rs

```diff
@@ -105,3 +107,32 @@ where
 
     Ok(query.all(db).await?)
 }
+
+/// Find all records of an entity by the given query filter map with optional sorting.
+pub async fn find_by_with_sort<E, C>(
```

Comment:
Would it be possible to add sorting to `find_by` above and apply sorting if the proper parameters, etc. are present? Seems like that's actually what this is doing so maybe we can just update `find_by` to look more like this? 

Ideally we would either encode sorting into `IntoQueryFilterMap` and use that trait instead of `CoachingSessionSortParams` above or we make the concept of sorting generic and have it be it's own trait within this module. I don't have a solid implementation in mind but that's the direction I hope we can go. Let me know if I can help at all with thinking through that or if it just straight up doesn't make sense.

Reply:
Done.